Tests/RoomMembershipTests.swift

```diff
@@ -0,0 +1,471 @@
+import XCTest
+import PusherPlatform
+@testable import PusherChatkit
+
+class RoomMembershipTests: XCTestCase {
+    var aliceChatManager: ChatManager!
+    var bobChatManager: ChatManager!
+    var roomId: Int!
+
+    override func setUp() {
+        super.setUp()
+
+        aliceChatManager = newTestChatManager(userId: "alice")
+        bobChatManager = newTestChatManager(userId: "bob")
+
+        let deleteResourcesEx = expectation(description: "delete resources")
+        let createRolesEx = expectation(description: "create roles")
+        let createAliceEx = expectation(description: "create Alice")
+        let createBobEx = expectation(description: "create Bob")
+
+        deleteInstanceResources() { err in
+            guard err == nil else {
+                fatalError(err!.localizedDescription)
+            }
+            deleteResourcesEx.fulfill()
+
+            createStandardInstanceRoles() { err in
+                guard err == nil else {
+                    fatalError(err!.localizedDescription)
+                }
+                createRolesEx.fulfill()
+            }
+
+            createUser(id: "alice") { err in
+                guard err == nil else {
+                    fatalError(err!.localizedDescription)
+                }
+                createAliceEx.fulfill()
+            }
+
+            createUser(id: "bob") { err in
+                guard err == nil else {
+                    fatalError(err!.localizedDescription)
+                }
+                createBobEx.fulfill()
+            }
+
+            sleep(1)
+        }
+
+        waitForExpectations(timeout: 10)
+    }
+
+    override func tearDown() {
+        aliceChatManager.disconnect()
+        aliceChatManager = nil
+        bobChatManager.disconnect()
+        bobChatManager = nil
+    }
+
+    // MARK: Chat manager delegate tests
+
+    func testChatManagerUserJoinedRoomHookWhenUserJoins() {
+        let userJoinedRoomHookEx = expectation(description: "user joined room hook called")
+        let bobJoinedRoomEx = expectation(description: "bob joined room")
+
+        let userJoinedRoom = { (room: PCRoom, user: PCUser) -> Void in
+            XCTAssertEqual(user.id, "bob")
+            XCTAssertEqual(room.name, "mushroom")
+            userJoinedRoomHookEx.fulfill()
+        }
+
+        let aliceCMDelegate = TestingChatManagerDelegate(userJoinedRoom: userJoinedRoom)
+
+        self.aliceChatManager.connect(delegate: aliceCMDelegate) { alice, err in
+            guard err == nil else {
+                XCTFail(err!.localizedDescription)
+                return
+            }
```

Comment:
is this better than `XCTAssertNil(err)`?

Reply:
Ha, good point. Probably not - I'll use `XCTAssertNil`. Nice and concise, thanks!